process/smartContract/processorV2/vmOutputAccountsProcessor.go

```diff
@@ -0,0 +1,313 @@
+package processorV2
+
+import (
+	"bytes"
+	"math/big"
+
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	"github.com/ElrondNetwork/elrond-go-core/data"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/state"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+)
+
+type VMOutputAccountsProcessor struct {
+	sc       *scProcessor
+	vmInput  *vmcommon.VMInput
+	vmOutput *vmcommon.VMOutput
+	tx       data.TransactionHandler
+	txHash   []byte
+
+	processOutputTransfers bool
+	processStorageUpdates  bool
+	processCode            bool
+	processBalanceDelta    bool
+	processNonce           bool
+}
+
+// NewVMOutputAccountsProcessor creates a new VMOutputProcessor instance.
+func NewVMOutputAccountsProcessor(
+	sc *scProcessor,
+	vmInput *vmcommon.VMInput,
+	vmOutput *vmcommon.VMOutput,
+	tx data.TransactionHandler,
+	txHash []byte) *VMOutputAccountsProcessor {
+	return &VMOutputAccountsProcessor{
+		sc:       sc,
+		vmInput:  vmInput,
+		vmOutput: vmOutput,
+		tx:       tx,
+		txHash:   txHash,
+
+		processOutputTransfers: false,
+		processStorageUpdates:  false,
+		processCode:            false,
+		processBalanceDelta:    false,
+		processNonce:           false,
+	}
+}
+
+// Run runs the configured processing steps on vm output
+func (oap *VMOutputAccountsProcessor) Run() (bool, []data.TransactionHandler, error) {
```

Comment:
hard to follow the flow. We have some functions that only set some paths and a function Run that can do all processing based on flags in any given combination. Why not split this in smaller functions and called them as required? One change here and you altered all flows that call this function with, maybe, unpredictable results. 
We do not follow this principle in the rest of the code.